### DISCUSSION TITLE
Locating dylib files

### DISCUSSION THREAD

**Discussion by tillkruss** [Homebrew/discussions#1592] [Tap maintenance and Homebrew development] [answered]
Is there a recommended way to locate `.dylib` files installed with Homebrew?

It seems like most are in `/opt/homebrew/lib/`:

```
/opt/homebrew/lib/libev.dylib
/opt/homebrew/lib/libjemalloc.dylib
/opt/homebrew/lib/liblz4.dylib
```

However OpenSSL doesn't seem to be present there:

```
/opt/homebrew/Cellar/openssl@1.1/1.1.1k/lib/libssl.dylib
/opt/homebrew/Cellar/openssl@1.1/1.1.1k/lib/libcrypto.dylib
```

We need to run `MachO::Tools.change_install_name` to update paths for a proprietary binary.

**Comment by carlocab** [comment, marked as answer]
If it's not keg-only (i.e. most formulae), then you can usually find them in `${HOMEBREW_PREFIX}/lib`. If it is keg-only (e.g. OpenSSL),  then you can (usually) find them at `${HOMEBREW_PREFIX}/opt/${formula_name}/lib` (e.g. `/opt/homebrew/opt/openssl@1.1/lib`).

You really want to avoid paths that contain `Cellar` -- these will change with version/revision bumps. This will break your dylib links even when the updated libraries are API/ABI-compatible. This is also why the install names of these libraries will not contain a reference to the `Cellar`.

That said, it's a little strange to need `MachO::Tools.change_install_name`. That's usually a sign that something's broken in the build system.

**Reply by tillkruss** [reply]
Thanks @carlocab, that exactly what I was looking for.

**Reply by tillkruss** [reply]
@carlocab: If I want to sign my binary during the build process, should I just preemptively point all dependencies at the default symlinks?

```
Formula["libev"].opt_lib/"libev.dylib"
Formula["openssl"].opt_lib/"libssl.dylib"
```
